src/tabs/AOCTab/days/day3/Day3Solution.js

```diff
@@ -1,14 +1,136 @@
 const Day3Solution = (props) => {
+    // Parses file, separating string into individual values
   const input = props.input
-  console.log("Day 3 input: " + input); 
+  var binary_input = input.split("\r\n");
+
+  // Lengths of binary digits and array
+  const width = binary_input[0].length; 
+  const height = binary_input.length;
+
+  let binary_input_gamma = "";
+  let binary_input_epsilon = "";
+
+  for (let c = 0; c < width; c++) {
+    /**
+     * Find most common bit in gamma rate
+     * Access each individual value in list
+     */
+    let count_0 = 0;
+    let count_1 = 0;
+    /**
+     * Calculates the number of times first bit of each number is present 
+     */ 
+    for (let r = 0; r < height; r++) {
+      parseInt(binary_input[r][c]) === 0 ? count_0 += 1 : count_1 += 1;
+      parseInt(binary_input[r][c]) === 1 ? count_1 += 1 : count_0 += 1;
+    }
+
+    /**
+     * Adds 0 or 1 to gamma and epsilon depending on which bit shows up more
+     */
+    count_0 > count_1 ? binary_input_gamma += "0" : binary_input_gamma += "1";
+    count_0 > count_1 ? binary_input_epsilon += "1" : binary_input_epsilon += "0";
+  }
+
+  /**
+   * Convert binary to integer
+   */
+  const gamma = parseInt(binary_input_gamma, 2);
+  const epsilon = parseInt(binary_input_epsilon, 2 );
+
+  const part1_answer = gamma * epsilon;
+ 
+ /********************************************************
+                        Part 2 
+ ********************************************************/ 
+  let bit_criteria = 0;
+  var row_p2 = binary_input.length;
+  let column_p2 = binary_input[0].length;
+  let binary_input_copy = [];
+  let answer = [];
+  
+  const generatorBitCriteria = (zero_bits, one_bits) => {
+    if (zero_bits > one_bits) {
+      bit_criteria = 0;

```

Comment:
OOOh okay so we've got what's called a "side effect" here, where we're updating a variable in the function that is neither in the inputs nor the outputs of the function. This is technically bad practice because it's really hard to know that this is going on without diving into the function. I think it's fine sometimes, but you should make it clear that this is going on in either the function name or the function comments